pure3.md

```diff
@@ -0,0 +1,225 @@
+validate that a provided SBOM is complete.
+
+## Specification
+
+Note: The reader is expected to be familiar with the in-toto specification.
+
+We propose in-toto as an option to achieve software supply chain security
+alongside the Uptane deployment. In such a model, in-toto can be backed by
+the Uptane root of trust, and use Uptane's delegation structure to distribute
+all of in-toto's metadata. Essentially, in-toto metadata is used to attest to
+the steps performed in the software supply chain itself, while Uptane is used
+to securely distribute that metadata alongside the ECU images. This combination
+of in-toto for software supply chain guarantees and Uptane for secure
+distribution of ECU images and in-toto metadata is called Scudo.
+
+### Modifications to the Repositories and Vehicle Interactions
+
+In this document, we propose updating one of Uptane's two repositories to store
+in-toto metadata alongside the images. The Image repository is recommended as
+its contents are typically signed for using offline keys, while the Director
+repository usually uses online keys.
+
+When an image is submitted to the Image repository, it must be accompanied by
+its in-toto metadata. The in-toto layout and link metadata for every image are
+noted in the opaque "custom" field of the image's entry in Targets metadata.
+This way, there is no ambiguity about the in-toto metadata to be used when an
+image is fetched and verified. Further, as it may not be practical for the top
+level Targets role to issue new metadata for every update, Uptane's delegations
+model can be used to make this entire process more scalable. Instead, delegated
+roles can be used to sign for the image itself and its in-toto link metadata.
+However, in-toto layouts and the key artifacts used to verify them should be
+signed for by the top level Targets role.
+
+Once the Director repository decides which images are required and signs Targets
+metadata for them, the vehicle must fetch the images and all the metadata using
+the interfaces provided by both the repositories. In this phase, the Director
+repository can continue signing targets metadata for the images, as per the
+default functioning of Uptane, and serve the metadata to the automobile. The
+automobile fetches the image itself from the Image repository along with its
+corresponding in-toto metadata.
+
+### Full Verification ECUs
+
+The Primary ECU receives images and metadata from the repositories, and sends
+them on to the other ECUs that it updates. Each Full Verification ECU is
+extended to perform independent in-toto verification that will be secure even
+if the Primary ECU is compromised. Some Full Verification ECUs, for example the
+Primary, are also expected to perform in-toto verification on behalf of Partial
+Verification ECUs that are resource constrained.
+
+For every target image the client needs to install, the corresponding in-toto
```

Comment:
You may wish to refer to the [datadog-checks-downloader](https://github.com/DataDog/integrations-core/blob/df7d87fe2888800bccb49c01a8cc5afa0f016e62/datadog_checks_downloader/datadog_checks/downloader/download.py#L282) for an example of a concrete client verification workflow.